src/cipher.rs

```diff
@@ -86,6 +87,30 @@ pub extern "C" fn rustls_supported_ciphersuite_get_suite(
     supported_ciphersuite.suite.get_u16()
 }
 
+/// Write the name of the ciphersuite into the provided buffer, up to a max of
+/// `len` bytes. Output is NUL terminated. Return the number of bytes written
+/// before the NUL.
+#[no_mangle]
+pub extern "C" fn rustls_supported_ciphersuite_get_name(
+    supported_ciphersuite: *const rustls_supported_ciphersuite,
+    buf: *mut c_char,
+    len: size_t,
+) -> size_t {
+    let write_buf: &mut [u8] = unsafe {
+        if buf.is_null() {
+            return 0;
+        }
+        slice::from_raw_parts_mut(buf as *mut u8, len as usize)
+    };
+    let supported_ciphersuite = try_ref_from_ptr!(supported_ciphersuite);
+    let suite_name: String = format!("{:?}", supported_ciphersuite);
+    let suite_name: &[u8] = suite_name.as_bytes();
+    let len: usize = min(write_buf.len() - 1, suite_name.len());
+    write_buf[..len].copy_from_slice(&suite_name[..len]);
```

Comment:
Since there is an implementation of `std::io::Write` for `&mut [u8]` I think we could write format directly into the user-provided buffer with `write_buf.write_fmt(format_args!("{:?}", supported_ciphersuite))`. Do you want to try that? The documentation at https://doc.rust-lang.org/std/io/trait.Write.html#impl-Write-6 seems to indicate this will correctly take care of the situation where the cipher suite name is longer than the provided buffer.

Reply:
> I think we could format directly into the user-provided buffer with write_buf.write_fmt(format_args!("{:?}", supported_ciphersuite)). Do you want to try that?

Sure. I copied this implementation from `rustls_version` - should I update that one as well?

Reply:
Sounds good, but maybe do it in a separate commit?

Reply:
I've now made this change